src/analysis/convex/cone.lean

```diff
@@ -185,6 +186,159 @@ ext' $ preimage_comp.symm
 @[simp] lemma mem_comap {f : E →ₗ[ℝ] F} {S : convex_cone F} {x : E} :
   x ∈ S.comap f ↔ f x ∈ S := iff.rfl
 
+/-- A convex cone is pointed if it includes 0 -/
+def pointed (S : convex_cone E) := (0 : E) ∈ S
+
+/-- A convex cone is blunt if it doesn't include 0 -/
+def blunt (S : convex_cone E) := (0 : E) ∉ S
+
+/-- A convex cone is flat if it contains some nonzero vector x and its opposite -x -/
+def flat (S : convex_cone E) := ∃ x ∈ S, x ≠ (0 : E) ∧ -x ∈ S
+
+/-- A convex cone is salient if it doesn't include x and -x for some nonzero x -/
+def salient (S : convex_cone E) := ∀ x ∈ S, x ≠ (0 : E) → -x ∉ S
+
+lemma pointed_iff_not_blunt (S : convex_cone E) : pointed S ↔ ¬blunt S :=
+  ⟨λ h₁ h₂, by finish, λ h, by finish⟩
```

Comment:
Maybe false.elim is far less expensive than finish?

Reply:
Yeah, that was a bit lazy on my part...! I've removed those.